lib/gitpassword.sh

```diff
@@ -1,41 +1,71 @@
-#!/bin/bash
+#!/bin/bash -eu
 
-if [ "$1" != "get" ]; then
-	echo "Usage: $0 get"
-	exit 1
-fi
-
-baseDir() {
-	# see https://stackoverflow.com/questions/59895
-	SOURCE=${BASH_SOURCE[0]}
-	while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
-		DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
-		SOURCE=$(readlink "$SOURCE")
-		[[ $SOURCE != /* ]] && SOURCE=$DIR/$SOURCE # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
-        done
-        DIR=$( cd -P "$( dirname "$SOURCE" )/.." >/dev/null 2>&1 && pwd )
-        echo $DIR
-}
+#echo "Called: $@" >> /tmp/credhelper
 
-BASE=$(baseDir)
-cd $BASE
+SECRETS_FILE=/var/cache/bridgehead/secrets/gitetcbridgehead
 
-source lib/functions.sh
+# Called from bridgehead, so shift 2 times
+shift
+shift
```

Comment:
Not that it really matters but would not shifting before calling this make more sense?

Reply:
Additionally: isn't `gitpassword.sh` also called from git as a credentials helper?

Reply:
No, as we need in this script some environment variables setup in the bridgehead script (SITE_ID etc.), we decided that it would make more sense to provide this script as part of the bridgehead script. The git credential helper will be configured like this: `credential.helper="/srv/docker/bridgehead/bridgehead gitCredentials <project>"`
Because of this the 2 times `shift` is necessary to pass the inputs of git calling the credential helper to the gitpassword.sh script.